nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/FetchInsightURI.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nuxeo
+ */
+package org.nuxeo.ai.model.serving;
+
+import org.nuxeo.ai.cloud.CloudClient;
+import org.nuxeo.ai.cloud.CloudConfigDescriptor;
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.core.api.DocumentNotFoundException;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Fetching Insight URI for the current user
+ */
+@Operation(id = FetchInsightURI.ID, category = Constants.CAT_DOCUMENT, label = "Fetching Insight URI", description = "Fetching Insight URI for the current user")
+public class FetchInsightURI {
+
+    public static final String ID = "AI.FetchInsightURI";
+
+    @OperationMethod
+    public String run() {
+        CloudClient cloudClient = Framework.getService(CloudClient.class);
+        CloudConfigDescriptor cloudConfig = cloudClient.getCloudConfig();
+        if (cloudConfig == null) {
+            throw new DocumentNotFoundException("Client Cloud configuration is missing");
```

Comment:
Are you sure that `DocumentNotFoundException` is the best here? Why not simple `OperationException` with your definitive message

Reply:
Cause i want to use the 404 not found code return.... Couldn't find a way to customize it. (some nuxeo documentation are not up to date)